diff.go

```diff
@@ -0,0 +1,228 @@
+		fmt.Println("---", *remoteSv.ServiceArn)
+		fmt.Println("+++", d.config.ServiceDefinitionPath)
+		fmt.Println(ds)
+	}
+
+	// task definition
+	newTd, err := d.LoadTaskDefinition(d.config.TaskDefinitionPath)
+	if err != nil {
+		return errors.Wrap(err, "failed to load task definition")
+	}
+
+	remoteTd, err := d.DescribeTaskDefinition(ctx, *newTd.Family)
+	if err != nil {
+		return errors.Wrap(err, "failed to describe task definition")
+	}
+
+	// sort lists in task definition
+	sortTaskDefinitionForDiff(newTd)
+	sortTaskDefinitionForDiff(remoteTd)
+
+	newTdBytes, err := MarshalJSON(tdToRegisterTaskDefinitionInput(newTd))
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal new task definition")
+	}
+
+	remoteTdBytes, err := MarshalJSON(tdToRegisterTaskDefinitionInput(remoteTd))
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal remote task definition")
+	}
+
+	if ds := diff.Diff(string(remoteTdBytes), string(newTdBytes)); ds != "" {
+		fmt.Println("---", *remoteTd.TaskDefinitionArn)
+		fmt.Println("+++", d.config.TaskDefinitionPath)
+		fmt.Println(ds)
+	}
+	return nil
+}
+
+func tdToRegisterTaskDefinitionInput(td *ecs.TaskDefinition) *ecs.RegisterTaskDefinitionInput {
+	return &ecs.RegisterTaskDefinitionInput{
+		ContainerDefinitions:    td.ContainerDefinitions,
+		Cpu:                     td.Cpu,
+		ExecutionRoleArn:        td.ExecutionRoleArn,
+		Family:                  td.Family,
+		Memory:                  td.Memory,
+		NetworkMode:             td.NetworkMode,
+		PlacementConstraints:    td.PlacementConstraints,
+		RequiresCompatibilities: td.RequiresCompatibilities,
+		TaskRoleArn:             td.TaskRoleArn,
+		ProxyConfiguration:      td.ProxyConfiguration,
+		Volumes:                 td.Volumes,
+	}
+}
+
+var stringerType = reflect.TypeOf((*fmt.Stringer)(nil)).Elem()
+
+func sortSlicesInDefinition(t reflect.Type, v reflect.Value, fieldNames ...string) {
+	isSortableField := func(name string) bool {
+		for _, n := range fieldNames {
+			if n == name {
+				return true
+			}
+		}
+		return false
+	}
+	for i := 0; i < t.NumField(); i++ {
+		fv, field := v.Field(i), t.Field(i)
+		if fv.Kind() != reflect.Slice || !fv.CanSet() {
+			continue
+		}
+		if !isSortableField(field.Name) {
+			continue
+		}
+		if size := fv.Len(); size == 0 {
+			fv.Set(reflect.MakeSlice(fv.Type(), 0, 0))
+		} else {
+			slice := make([]reflect.Value, size, size)
+			for i := 0; i < size; i++ {
+				slice[i] = fv.Index(i)
+			}
+			sort.Slice(slice, func(i, j int) bool {
+				iv, jv := reflect.Indirect(slice[i]), reflect.Indirect(slice[j])
+				var is, js string
+				if iv.Kind() == reflect.String && jv.Kind() == reflect.String {
+					is, js = iv.Interface().(string), jv.Interface().(string)
+				} else if iv.Type().Implements(stringerType) && jv.Type().Implements(stringerType) {
+					is, js = iv.Interface().(fmt.Stringer).String(), jv.Interface().(fmt.Stringer).String()
+				}
+				return strings.Compare(is, js) < 0
```

Comment:
You can write it simpler.

```suggestion
				return is < js
```